client/dashboard/src/pages/logs/Logs.tsx

```diff
@@ -0,0 +1,532 @@
+import { Page } from "@/components/page-layout";
+import { Badge } from "@/components/ui/badge";
+import { SearchBar } from "@/components/ui/search-bar";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Sheet,
+  SheetContent,
+} from "@/components/ui/sheet";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { useListToolLogs } from "@gram/client/react-query";
+import { HTTPToolLog } from "@gram/client/models/components";
+import { Icon } from "@speakeasy-api/moonshine";
+import { Copy, ExternalLink, FileCode, SquareFunction } from "lucide-react";
+import { useState, useRef, useEffect } from "react";
+
+function StatusIcon({ isSuccess }: { isSuccess: boolean }) {
+  if (isSuccess) {
+    return (
+      <div style={{ color: "var(--fill-success-default, #5a8250)" }}>
+        <Icon name="check" className="items-center size-4" />
+      </div>
+    );
+  }
+  return (
+    <div style={{ color: "var(--fill-destructive-default, #c83228)" }}>
+      <Icon name="x" className="size-4" />
+    </div>
+  );
+}
+
+export default function Logs() {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [toolTypeFilter, setToolTypeFilter] = useState<string>("");
+  const [serverNameFilter, setServerNameFilter] = useState<string>("");
+  const [statusFilter, setStatusFilter] = useState<string>("");
+  const [selectedLog, setSelectedLog] = useState<HTTPToolLog | null>(null);
+
+  // Infinite scroll state
+  const [allLogs, setAllLogs] = useState<HTTPToolLog[]>([]);
+  const [currentCursor, setCurrentCursor] = useState<string | undefined>(undefined);
+  const [isFetchingMore, setIsFetchingMore] = useState(false);
+  const lastProcessedCursorRef = useRef<string | undefined>(undefined);
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const perPage = 50;
+
+  const { data, isLoading } = useListToolLogs(
```

Comment:
I think there should be a `useListToolLogsInfinite` that speakeasy is supposed to generate that should handle most of this stuff for you, but it is vexingly not here. I'm gonna look into that